lib/ansible/modules/windows/win_share.ps1

```diff
@@ -111,26 +111,34 @@ Function NormalizeAccounts
 
 $result = @{
     changed = $false
+    actions = @() # More for debug purposes
 }
 
-$params = Parse-Args $args
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
 $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent"
 
+if (-not (Get-Command -Name Get-SmbShare -ErrorAction SilentlyContinue)) {
+    Fail-Json $result "The current host does not support the -SmbShare cmdlets required by this module. Please run on Server 2012 or Windows 8 and later"
+}
+
 Try {
-    $share = Get-SmbShare $name -ErrorAction SilentlyContinue
+    $share = Get-SmbShare -Name $name -ErrorAction SilentlyContinue
     If ($state -eq "absent") {
         If ($share) {
-            Remove-SmbShare -Force -Name $name
+            if (-not $check_mode) {
+                Remove-SmbShare -Force -Name $name
```

Comment:
You would think so but all the `*-SmbShare` cmdlets don't seem to honour it. Try it for yourself and see it actually creating/modifying/removing shared even when set to true.